### DISCUSSION TITLE
Redis server closes Jedis connection with ssl

### DISCUSSION THREAD

**Discussion by alaajel** [xetorthio/jedis#3193] [Q&A] [answered]
currently trying to get ssl connection working with Jedis, have setup everything including trust store etc, but when using this code
```
// based on Jedis test classes
Jedis jedis = new Jedis("localhost", 6379, true);
jedis.ping();
jedis.close(); 
```
which throws `JedisConnectionException` 

*this not full error*
```
Caused by: java.net.SocketException: Connection reset
	at sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:323) ~[?:?]
	at sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350) ~[?:?]
	at sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803) ~[?:?]
	at java.net.Socket$SocketInputStream.read(Socket.java:966) ~[?:?]
	at java.io.InputStream.read(InputStream.java:218) ~[?:?]
```
but looking at redis server console this was logged:
`Error accepting a client connection: error:1408F10B:SSL routines:ssl3_get_record:wrong version number`

i have made sure to test redis tls connection using `redis-cli` which worked without a problem.

not sure if this problem with way how i setup things. any answers for this?

**Comment by zeekling** [comment, marked as answer]
client use ssl3,but redis server not support ssl3,maybe support ssl2.
you can check this config of redis server:
![image](https://user-images.githubusercontent.com/17929800/200106541-7cdf2e5b-d8cd-472f-909b-fb62acc2202c.png)

**Reply by alaajel** [reply]
~~i am currently have it as start args `--tls-protocols "TLSv1.2 TLSv1.3"`
but still that same log is logged in redis console.~~
